README.md

```diff
@@ -22,3 +22,35 @@ or using a browser: `http://localhost:9090`
 
 Please refer to the [lorax-composer](https://github.com/weldr/lorax)'s documenation as osbuild-composer is a drop-in replacement.
 
+## High-level overview
+
+![overview](docs/osbuild-composer.svg)
+
+### Frontends
+
+`osbuild-composer` is meant to be used with 3 different front-ends. The primary one, which is meant for general use, is cockpit-composer. It is part of the Cockpit project and unless you have a strong reason not to use it, you should use it. `composer-cli` is a command line tool that can be used for non-interactive usage. Finally, RCM client (TBD) is a special-purpose client that is not meant for general usage.
```

Comment:
`composer-cli` is *not* for non-interactive usage. It is for humans and makes no guarantees about its output. All non-interactive usage outside of tests that are tightly coupled to the code repository should be using the API directly.

Reply:
thanks, I didn't know that